### PR TITLE
Added Permission checks #111

### PR DESCRIPTION
**Pull request by lxatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-WebServices-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-SDK [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-WebServices-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
Added Permission checks for File.IsLocked, DomainFileStore.IsHidden and ownership of DomainFileStore, File and Folder.

### CODE REVIEW

File: CDP4WebServices.API/Services/Supplemental/DomainFileStoreService.cs [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    /// </summary>
+    public sealed partial class DomainFileStoreService
+    {
+        /// <summary>
+        /// Gets or sets the <see cref="IIterationService"/>.
+        /// </summary>
+        public IIterationService IterationService { get; set; }
+
+        /// <summary>
+        ///
```
Comment (alexatrhea) [inline CDP4WebServices.API/Services/Supplemental/DomainFileStoreService.cs:51]: Docs

File: CDP4WebServices.API/Services/Supplemental/DomainFileStoreService.cs [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        public IIterationService IterationService { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        private readonly Dictionary<Type, Func<Guid, Func<DomainFileStore, bool>>> domainFileStoreSelectors = new Dictionary<Type, Func<Guid, Func<DomainFileStore, bool>>>
+        {
+            {
+                typeof(File),
+                guid =>
```
Comment (alexatrhea) [inline CDP4WebServices.API/Services/Supplemental/DomainFileStoreService.cs:57]: Interesting notation, I dont think I have seen this before.what does it mean exactly?

File: CDP4WebServices.API/Services/Supplemental/DomainFileStoreService.cs [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        /// Checks if the <see cref="Participant"/> is allowed to read (and therefore also write to) a <see cref="DomainFileStore"/>
+        /// based on the state of the <see cref="DomainFileStore"/>'s <see cref="DomainFileStore.IsHidden"/> property.
+        /// </summary>
+        /// <param name="transaction">
+        /// The current transaction to the database.
+        /// </param>
+        /// <param name="thing">
+        /// The <see cref="Thing"/> to check.
+        /// </param>
+        /// <returns></returns>
```
Comment (alexatrhea) [inline CDP4WebServices.API/Services/Supplemental/DomainFileStoreService.cs:163]: return docs

File: CDP4WebServices.API/Services/Supplemental/DomainFileStoreService.cs (hunk 2 above)
Comment (lxatrhea) [reply CDP4WebServices.API/Services/Supplemental/DomainFileStoreService.cs:57]: It's a Dictionary with a key of 'Type' and its value is a Func<> that returns a Func<>
Its usage was not yet implemented correctly though.
Fixed that, and I'll add a test for it.

**Pull request merged**